prober/utils_test.go

```diff
@@ -162,7 +162,7 @@ func TestChooseProtocol(t *testing.T) {
 	registry = prometheus.NewPedanticRegistry()
 
 	ip, _, err = chooseProtocol(ctx, "ip4", false, "ipv6.google.com", registry, logger)
-	if err != nil && err.Error() != "address ipv6.google.com: no suitable address found" {
+	if err != nil && err.Error() != "address ipv6.google.com: no suitable address found" && err.Error() != "lookup ipv6.google.com: no such host" {
```

Comment:
I don't understand this? can you please explain why we need this check?

Reply:
Was a fix for go 1.17 as the error message changed, #864 covers it too, so will rebase it out.

Reply:
I actually found a better fix.